website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+---
+layout: "guides"
+page_title: "Securing Nomad with TLS"
+sidebar_current: "guides-tls"
+description: |-
+  Securing Nomad's cluster communication is not only important for security but
+  can even ease operations by preventing mistakes and misconfigurations. Nomad
+  optionally uses mutual TLS (mTLS) for all HTTP and RPC communication.
+---
+
+# Securing Nomad with TLS
+
+Securing Nomad's cluster communication is not only important for security but
+can even ease operations by preventing mistakes and misconfigurations. Nomad
+optionally uses mutual TLS (mTLS) for all HTTP and RPC communication. Nomad's
+use of mTLS provides the following properties:
+
+* Prevent unauthorized Nomad access
+* Prevent observing or tampering with Nomad communication
+* Prevent client/server role or region misconfigurations
+
+The 3rd property is fairly unique to Nomad's use of TLS. While most uses of TLS
+verify the identity of the server you're connecting to based on a domain name
+such as `nomadproject.io`, Nomad verifies the node you're connecting to is in
+the expected region and configured for the expected role (e.g.
+`client.us-west.nomad`).
+
+Configuring TLS can be unfortunately complex process, but if you used the
+[Getting Started Guide's Vagrantfile][Vagrantfile] or have [cfssl][] and Nomad
+installed this guide will provide you with a production ready TLS
+configuration.
+
+~> Note that while Nomad's TLS configuration will be production ready, key
+   management and rotation is a complex subject not covered by this guide.
+   [Vault][] is the suggested solution for key generation and management.
+
+## Creating Certificates
+
+The first step to configuring TLS for Nomad is generating certificates. In
+order to prevent unauthorized cluster access, Nomad requires all certificates
+be signed by the same Certificate Authority (CA). This should be a *private* CA
+and not a public one like [Let's Encrypt][letsencrypt] as any certificate
+signed by this CA will be allowed to communicate with the cluster.
+
+### Certificate Authority
+
+There are a variety of tools for managing your own CA, [like the PKI secret
+backend in Vault][vault-pki], but for the sake of simplicity in this guide
+we'll use [cfssl][]. You can generate a private CA certificate and key with
```

Comment:
I covered this above and am open to changing but I should probably clarify: I am not as confident in my ability to deliver correct instructions with OpenSSL as with cfssl. Definitely not as concise.

As to "in the wild" examples of cfssl, while OpenSSL clearly dominates, cfssl is quite popular in related products' documentation:

https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/02-certificate-authority.md
https://coreos.com/os/docs/latest/generate-self-signed-certificates.html
https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/
https://github.com/kelseyhightower/docker-kubernetes-tls-guide
https://www.digitalocean.com/community/tutorials/how-to-secure-your-coreos-cluster-with-tls-ssl-and-firewall-rules

coreos & k8s may have done it because cfssl is new and shiny, but I suspect they also need a lot of the features (SANs) we do that OpenSSL makes difficult.